crates/eth-rpc/src/main.rs

```diff
@@ -17,7 +21,11 @@ async fn main() -> Result<()> {
     let starknet_config = StarknetConfig::from_env()?;
     let rpc_config = RPCConfig::from_env()?;
 
-    let kakarot_rpc = KakarotRpc::new(starknet_config);
+    let provider = JsonRpcClientBuilder::with_http(&starknet_config).unwrap().build();
```

Comment:
no need to unwrap here I think

Reply:
ah well this was done before^^ interesting. 
i think we should utilize `eyre` to exit gracefully rather than unwrap

Reply:
if not, we can also use "expect"

Reply:
can be done in another issue, let's just create the issue and tackle this in another PR
great job

Reply:
Gotcha! So that makes two issues we have to create:
- Implement all Alchemy endpoints { low priority }
- Refactor to avoid unwrapping here.

After the merge if you want I can create them, or if you want to then that is also fine.

Reply:
Yes! you can create them sounds good